### PR TITLE
[ISSUE #367] Support prototype spi instance

### PR DESCRIPTION
**Pull request by ruanwenjun**
Change producer and consumer spi to prototype

### CODE REVIEW

File: eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java [hunk 1]
```diff
@@ -17,13 +17,24 @@
...
+import org.junit.Assert;
 import org.junit.Test;
 
 public class EventMeshExtensionFactoryTest {
 
     @Test
-    public void getExtension() {
-        TestExtension extensionA = EventMeshExtensionFactory.getExtension(TestExtension.class, "extensionA");
-        extensionA.hello();
+    public void getSingletonExtensionTest() {
```
Comment (dongzl) [inline eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java:28]: Hi @ruanwenjun , I think the `testGetSingleExtension` is better name. the prefix `testXXX` is better.

File: eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java [hunk 2]
```diff
@@ -17,13 +17,24 @@
...
+        TestSingletonExtension extensionA = EventMeshExtensionFactory.getExtension(TestSingletonExtension.class, "singletonExtension");
+        TestSingletonExtension extensionB = EventMeshExtensionFactory.getExtension(TestSingletonExtension.class, "singletonExtension");
+        Assert.assertSame(extensionA, extensionB);
+    }
+
+    @Test
+    public void getPrototypeExtensionTest() {
```
Comment (dongzl) [inline eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java:35]: The save to above.

File: eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java (hunk 1 above)
Comment (ruanwenjun) [reply eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java:28]: Yes, you are right. I will fix tonight.

Comment (ruanwenjun) [reply eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java]: done

File: eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java (hunk 2 above)
Comment (ruanwenjun) [reply eventmesh-spi/src/test/java/org/apache/eventmesh/spi/EventMeshExtensionFactoryTest.java]: done

**Pull request merged**